packages/workflow-diagram/src/layout/factories.ts

```diff
@@ -56,10 +74,10 @@ export function triggerNodeFactory(job: Job, workflow: Workflow): FlowElkNode {
     __flowProps__: {
       data: {
         label: TriggerLabels[job.trigger.type],
-        description: job.trigger.description,
-        workflow,
+        description: generateDescription(job.trigger),
+        workflow: renameWorkflowForUntitled(workflow),
```

Comment:
Ah man! So close! What I intended was that the actual Job/Workflow/Trigger object is passed in and the `generateDescription` would be in `TriggerWorkflowNode.tsx` for example. If we pass in the exact object we used to decide where the node goes in the tree we will be able to do all sorts of ah-hoc rendering in the component.

In the near future we might allow us to inject the nodes just before using the library, meaning we might have a moment when we have this specific nodes inside Lightning.

Reply:
Hey @stuartc this is done now. Thanks for the catch. Seems much better now. Give it another round of review and let me know.

Reply:
Done